### PR TITLE
[SDK-921] call to update dimensions on stream when window or viewer element is resized

### PR DESCRIPTION
**Pull request by danconner-vertex**
## What

Send 'update dimensions' on the stream when the viewer element has been resized due to browser re-sizing or other.

## Ticket

https://vertexvis.atlassian.net/browse/SDK-921

## Test Plan

Force a resize of the viewer element both by resizing the browser window, and by javascript updating styles to it's resized.

Via javascript, in the console:
```
document.getElementById('viewer').setAttribute('style', 'width: 90%')
```
to set back:
```
document.getElementById('viewer').setAttribute('style', '')
```

In both cases, verify new frame with proper dimensions is rendered.

## Areas of Possible Regression

n/a

## Out of scope changes made in PR

n/a

## Dependencies

https://github.com/Vertexvis/frame-streaming-service/pull/92

### CODE REVIEW

File: packages/viewer/package.json [hunk 1]
```diff
@@ -45,12 +45,13 @@
...
   "dependencies": {
     "@types/classnames": "2.2.3",
-    "@vertexvis/frame-streaming-protos": "^0.1.4",
+    "@vertexvis/frame-streaming-protos": "^0.1.5",
     "@vertexvis/geometry": "0.9.0",
     "@vertexvis/stream-api": "0.9.0",
     "@vertexvis/utils": "0.9.0",
     "classnames": "2.2.6",
     "protobufjs": "^6.9.0",
+    "resize-observer-polyfill": "^1.5.1",
```
Comment (danconner-vertex) [inline packages/viewer/package.json:54]: types coming to typescript soon: https://github.com/microsoft/TSJS-lib-generator/issues/908

wide support in the latest versions of modern browsers, but nothing in IE

File: packages/stream-api/src/streamApi.ts [hunk 2]
```diff
@@ -160,6 +161,30 @@ export class StreamApi {
...
+   *
+   * @param payload The payload of the request.
+   * @param withResponse Indicates if the server should reply with a response.
+   * Defaults to `true`.
+   */
+  public updateDimensions(
+    payload: UpdateDimensionsPayload,
+    withResponse = true
+  ): Promise<vertexvis.protobuf.stream.IStreamResponse> {
+    console.log('dimensions', payload);
```
Comment (MadisonEhlers-Vertex) [inline packages/stream-api/src/streamApi.ts:184]: nit: should remove before merging.

**Review by MadisonEhlers-Vertex — APPROVED** [review]

File: packages/viewer/src/components/viewer/viewer.tsx [hunk 3]
```diff
@@ -429,14 +437,22 @@ export class Viewer {
...
-  private handleWindowResize(event: UIEvent): void {
+  private handleResize(): void {
     if (!this.isResizing) {
       this.isResizing = true;
 
       window.requestAnimationFrame(() => this.recalculateComponentDimensions());
     }
   }
 
+  private handleWindowResize(event: UIEvent): void {
```
Comment (danschultz) [inline packages/viewer/src/components/viewer/viewer.tsx:448]: Do we still need this?

File: packages/viewer/src/components/viewer/viewer.tsx [hunk 4]
```diff
@@ -493,8 +509,7 @@ export class Viewer {
       this.calculateComponentDimensions();
       this.isResizing = false;
 
-      // TODO(dan): Need to add and invoke message for resizing the image
-      // stream. https://vertexvis.atlassian.net/browse/SDK-921
+      this.stream.updateDimensions({ dimensions: this.dimensions });
```
Comment (danschultz) [inline packages/viewer/src/components/viewer/viewer.tsx:512]: Will this throw errors if the WS connection hasn’t been established?

Comment (danconner-vertex) [reply packages/viewer/src/components/viewer/viewer.tsx:512]: probably so. is there now a way to check if the connection has been established and 'connected'?

File: packages/viewer/src/components/viewer/viewer.tsx (hunk 3 above)
Comment (danconner-vertex) [reply packages/viewer/src/components/viewer/viewer.tsx:448]: good call, shouldn't be.

File: packages/viewer/src/components/viewer/viewer.tsx (hunk 4 above)
Comment (danschultz) [reply packages/viewer/src/components/viewer/viewer.tsx:512]: would probably be nice to have a helper function in the `Viewer` component or `StreamApi`, but you can check the existence of `streamDisposable`.

Comment (danconner-vertex) [reply packages/viewer/src/components/viewer/viewer.tsx:509]: updated in fcddbcf289ccdc6276ad59d7afa8c0ec7521b105

**Review by MadisonEhlers-Vertex — APPROVED** [review]

**Pull request merged**